transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -1148,11 +1178,16 @@ public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
         private final Unsafe unsafe;
 
         HeadContext(DefaultChannelPipeline pipeline) {
-            super(pipeline, null, HEAD_NAME, HeadContext.class);
+            super(pipeline, HEAD_NAME, HeadContext.class);
             unsafe = pipeline.channel().unsafe();
             setAddComplete();
         }
 
+        @Override
+        public EventExecutor executor() {
+            return channel.eventLoop();
```

Comment:
@normanmaurer follow on from prior comment here ... I wonder if it's a problem though that the executor is different for the head context, given we now rely on the fact that all the changes to next/prev pointers in `AbstractChannelHandlerContext` happen in the same (ordered/serialized) executor?

Reply:
hmm yeah.... Let me just only support the Eventloop for the whole pipeline. If we really need the feature later on we can do a followup.